IDE/iotsafe-raspberrypi/README.md

```diff
@@ -0,0 +1,142 @@
+
+| File Slot | Name                 | Description                                                |
+| --------- | -------------------- | ---------------------------------------------------------- |
+| 0x03      | `CRT_SERVER_FILE_ID` | pre-provisioned with the server ECC public key certificate |
+
+How the applet configuration (such as putting the client key pair with the corresponding client certificate and the server certificate the client can trust) is performed during its initial and on-field lifetime depends on the infrastructure and is out-of-scope of this demo.
+
+
+### IoT-SAFE Interface
+
+The below code explanations can be found in the `client-tls13.c`.
+
+In this demo, the client is the IoT-SAFE capable endpoint.
+
+First, it creates a wolfSSL context `ctx` with TLS 1.3.
+
+```c
+wolfSSL_CTX_new(wolfTLSv1_3_client_method());
+```
+
+In order to activate IoT-SAFE support in this context, the following function is
+called:
+
+```c
+wolfSSL_CTX_iotsafe_enable(ctx);
+```
+
+Extracting the client and server certificate can be done by the following functions:
+
+```c
+wolfIoTSafe_GetCert(
+                    CRT_CLIENT_FILE_ID,
+                    cert_buffer,
+                    sizeof(cert_buffer));
+```
+
+```c
+wolfIoTSafe_GetCert(
+                    CRT_SERVER_FILE_ID,
+                    cert_buffer,
+                    sizeof(cert_buffer));
+```
+
+in which the extracted certificate inside the `cer_buffer` can be later loaded to the `ctx`.
+
+Additionally, after the TLS session `ssl` creation, shown below:
+
+```c
+ssl = wolfSSL_new(ctx);
+```
+
+the client associates the pre-provisioned keys and the available slots in the
+IoT-SAFE applet to the current session:
+
+
+```c
+wolfSSL_iotsafe_on(ssl, PRIVKEY_ID, ECDH_KEYPAIR_ID, PEER_PUBKEY_ID, PEER_CERT_ID);
+```
+
+
+### Compiling
+
+First, user needs to build wolfSSL with the following options:
+```
+./configure CFLAGS="-DWOLFSSL_TRUST_PEER_CERT" --enable-tls13 --enable-pkcallbacks --enable-debug --enable-iotsafe --enable-hkdf
+```
+
+Additionally, user can pass `CFLAGS="-DDEBUG_WOLFSSL -DWOLFSSL_DEBUG_TLS -DDEBUG_IOTSAFE"` if more debugging information is to be used. This can clutter the demo stdout more than `--enable-debug` does, but this is very useful to see the overall TLS 1.3 handshaking process with IoT-SAFE.
+
+Hence, the full wolfSSL build for the demo is:
+```
+./configure CFLAGS="-DWOLFSSL_TRUST_PEER_CERT -DDEBUG_WOLFSSL -DWOLFSSL_DEBUG_TLS -DDEBUG_IOTSAFE" --enable-tls13 --enable-pkcallbacks --enable-debug --enable-iotsafe
+```
+
+`-DWOLFSSL_TRUST_PEER_CERT` is needed for `wolfSSL_CTX_trust_peer_buffer` in `IDE/iotsafe-raspberrypi/client-tls13.c`
+
+### Running
+
+After building wolfSSL, from this directory, run `make` and a help usage will be shown.
```

Comment:
At this point, `README.md` refers to a local Makefile, which is not among the files added to this PR.